pdl_syntax.md

```diff
@@ -0,0 +1,828 @@
+
+
+### Deprecation
+Types and fields can be deprecated by adding @deprecated annotation.
+
+For example :
+
+```
+@deprecated = "Use record X instead."
+record Example {
+  @deprecated = "Use field x instead."
+  field: string
+}
+```
+
+### Including fields
+
+Record can include fields from other record.
+
+For example : 
+
+```
+namespace com.linkedin.pegasus.generator.examples
+
+/**
+ * Bar includes fields of Foo, Bar will have fields f1 from itself and b1 from Bar
+ */
+record Bar includes Foo {
+  b1: string
+}
+```
+```
+namespace com.linkedin.pegasus.generator.examples
+
+record Foo {
+  f1: string
+}
+```
+In pegasus, field inclusion does not imply inheritance, it is merely a convenience to reduce duplication when writing schemas.
+
+
+### Properties
+Properties can be used to present arbitrary datas and added to records, fields and enums.
+
+For example : 
+```
+@prop = "value"
+record Fruits {
+  @validate.regex.regex = "^(yes|no)$"
+  field: string
+}
+```
+#### Property values can be any valid JSON type:
+
+For example : 
+```
+@prop = 1
+```
+```
+@prop = "string"
+```
+```
+@prop = [1, 2, 3]
+```
+```
+@prop = { "a": 1", "b": { "c": true }}
+```
+
+#### Property keys can expressed as paths:
+
+For example : 
+```
+@java.class = "org.joda.DateTime"
+@java.coercerClass = "org.example.DateTimeCoercer"
+```
+
+The dot seperate keys format is equivalent to JSON value format.
+
+The above dot seperate keys example is equivalent to the following JSON value example :
+```
+@java = {
+  "class": "org.joda.DateTime",
+  "coercerClass": "org.example.DateTimeCoercer"
+}
+```
+
+### Escaping
+There are some keywords which are reserved in Pegasus. If you have to use them to define any names, you need to put them in backticks : ` `.
+
+#### Keyword Escaping
+
+```
+namespace com.linkedin.pegasus.generator.test.idl.escaping
+
+record PdlKeywordEscaping {
+  `namespace`: string
+  `record`: string
+  `null`: string
+  `enum`: string
+  recordName: record `record` { }
+}
+```
+
+#### Namespace/Package escaping
+
+```
+namespace com.linkedin.pegasus.generator.test.idl.escaping.`record`
+package com.linkedin.pegasus.generator.test.idl.escaping.override.`typeref`
+
+/**
+ * Ensures that the namespace and package are properly escaped at the root as well as in scoped named-type declarations.
+ */
+record NamespacePackageEscaping {
+  x: {
+    namespace com.x.y.z.`enum`
+    package com.a.b.c.`fixed`
+
+    record Foo {}
+  }
+}
+```
+
+#### Property key escaping 
+
+If you want Pegasus to treat property key name with dots as one string key, please use backticks to escape such string.
+For example : 
+
+```
+namespace com.linkedin.pegasus.generator.test.idl.escaping
+
+record PropertyKeyEscaping {
+  @`namespace` = "foo.bar"
+  @`test.path` = 1
+  @validate.`com.linkedin.CustomValidator` = "foo"
+  aField: string
+}
+```
+
+
+## Primitive Types
+
+The Pegasus primitive types are : int, long, float, double, boolean, string and bytes.
```

Comment:
Include example for specifying default values

Reply:
Added an example for specifying default values.